src/atlas/grid/detail/distribution/DistributionImpl.h

```diff
@@ -44,15 +50,21 @@ class DistributionImpl : public util::Object {
 
     virtual ~DistributionImpl();
 
-    int partition( const gidx_t gidx ) const { return part_[gidx]; }
+    int partition( const gidx_t gidx ) const 
+    { 
+      if (part_.size () > 0)
+        return part_[gidx];
+      idx_t iblock = gidx / blocksize_;
+      return (iblock * nb_partitions_) / nb_blocks_;
+    }
```

Comment:
How important is inlining here?  I want to know if it could it be made virtual perhaps, preventing this to be inlined.

Reply:
Hello Willem,

I have just tested it on my reference test case (40000x20000), and it costs only a few seconds, so yes, you should make it virtual.

Philippe

Reply:
Sometimes seconds matter... But typically this is only at a setup though, but could matter even there for pre- and post- processing programs.
If you believe it is OK, then that opens up a lot of options, and the "if" could also be removed.

Reply:
Of course, I understand, please go on.